sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChunkFactory.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using Azure.Storage.Internal.Avro;
+
+namespace Azure.Storage.Blobs.ChangeFeed
+{
+    internal class ChunkFactory
+    {
+        private readonly LazyLoadingBlobStreamFactory _lazyLoadingBlobStreamFactory;
+        private readonly AvroReaderFactory _avroReaderFactory;
+
+        public ChunkFactory(
+            LazyLoadingBlobStreamFactory lazyLoadingBlobStreamFactory,
+            AvroReaderFactory avroReaderFactory)
+        {
+            _lazyLoadingBlobStreamFactory = lazyLoadingBlobStreamFactory;
+            _avroReaderFactory = avroReaderFactory;
+        }
+
+        public virtual Chunk BuildChunk(
+            BlobContainerClient containerClient,
```

Comment:
The client doesn't change from one `BuildChunk` invocation to another. So, this seem more like a dependency here rather than method input. I'd suggest to refactor these factories in such a way that their are created with dependencies early in the top level layer.
Corresponding refactoring in Java https://github.com/Azure/azure-sdk-for-java/pull/10839/commits/6b3fa618d06b18c678e2ec3eb6bcb3a53831d7cf

Reply:
Made containerClient a dependency.